internal/cluster/query.go

```diff
@@ -2,53 +2,58 @@ package cluster
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/buildkite/cli/v3/internal/graphql"
 	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
 	"golang.org/x/sync/errgroup"
 )
 
-func QueryCluster(ctx context.Context, OrganizationSlug string, ClusterID string, f *factory.Factory) (*Cluster, error) {
-	q, err := graphql.GetClusterQueues(ctx, f.GraphQLClient, OrganizationSlug, ClusterID)
+func GetQueues(ctx context.Context, f *factory.Factory, orgSlug string, clusterID string, lo *buildkite.ClusterQueuesListOptions) ([]ClusterQueue, error) {
+	queues, _, err := f.RestAPIClient.ClusterQueues.List(orgSlug, clusterID, lo)
 	if err != nil {
-		return nil, fmt.Errorf("unable to read Cluster Queues: %s", err.Error())
-	}
-
-	ClusterDescription := q.Organization.Cluster.Description
-	cluster := Cluster{
-		OrganizationSlug: OrganizationSlug,
-		ClusterID:        ClusterID,
-		Name:             q.Organization.Cluster.Name,
-		Description:      string(*ClusterDescription),
-		Queues:           make([]Queue, len(q.Organization.Cluster.Queues.Edges)),
+		return nil, err
 	}
 
-	eg, ctx := errgroup.WithContext(ctx)
-
-	for i, edge := range q.Organization.Cluster.Queues.Edges {
-
-		i, edge := i, edge
+	eg, _ := errgroup.WithContext(ctx)
+	queuesResponse := make([]ClusterQueue, len(queues))
+	for i, q := range queues {
+		i, q := i, q
```

Comment:
FYI I dont think this is necessary any more on our current version of Go right? Doesn't hurt though